src/transaction/log_manager.c

```diff
@@ -6113,6 +6171,67 @@ log_dump_record_commit_postpone (THREAD_ENTRY * thread_p, FILE * out_fp, LOG_LSA
   return log_page_p;
 }
 
+// *INDENT-OFF*
+void log_unpack_group_commit (THREAD_ENTRY * thread_p, LOG_LSA * log_lsa, LOG_PAGE * log_page_p, int buf_size,
+			      std::vector<rv_gc_info> & group)
+// *INDENT-ON*
+
+{
+  char *start_buf = log_page_p->area + log_lsa->offset;
+  bool needs_free = false;
+
+  if (log_lsa->offset + buf_size > (int) LOGAREA_SIZE)
+    {
+      /* Need to copy the data into a contiguous area */
+      start_buf = (char *) malloc (buf_size);
+      if (start_buf == NULL)
+	{
+	  logpb_fatal_error (thread_p, true, ARG_FILE_LINE, "log_unpack_group_commit");
+	  return;
+	}
+      needs_free = true;
+
+      /* Copy the data */
+      logpb_copy_from_log (thread_p, start_buf, buf_size, log_lsa, log_page_p);
+    }
+
+  const char *crt_buf = start_buf;
+  const char *end_of_buf = start_buf + buf_size;
+
+  while (crt_buf < end_of_buf)
+    {
+      TRANID trid;
+      TRAN_STATE state;
+      LOG_LSA pp_lsa;
+
+      LSA_SET_NULL (&pp_lsa);
+
+      assert (crt_buf + sizeof (trid) + sizeof (state) <= end_of_buf);
+
+      trid = *((TRANID *) crt_buf);
```

Comment:
I think the current code works and it might be better to add safe guards to read `TRANID`, `TRAN_STATE` and `LOG_LSA` which check they are properly aligned.

Reply:
added alignment asserts